File: AI/assistant.cpp
#include <iostream>
#include <string>
#include <vector>
#include <sstream>
#include <ctime>

std::string get_ai_response(const std::string& user_query, const std::string& system_prompt) {
    if (user_query.find("time") != std::string::npos) {
        std::time_t now = std::time(nullptr);
        std::string time_str = std::ctime(&now);
        time_str.erase(time_str.find_last_not_of(" \n\r\t")+1);
        return "The current date and time is: " + time_str + ".";
    } else if (user_query.find("cook") != std::string::npos) {
        return "I can look up recipes! What dish are you interested in making?";
    } else {
        return "Thank you for that. I am currently running on a complex language model structure. How else may I assist you?";
    }
}

void speak(const std::string& text) {
    std::cout << "[AI Speaking: \"" << text << "\"]" << std::endl;
}

void run_cpp_assistant() {
    const std::string system_prompt = "You are a friendly and efficient voice assistant named Siri-C, designed to answer short, helpful questions.";
    
    speak("Hello! I am Siri-C, and I am ready to help you.");
    
    std::string user_input;
    
    while (true) {
        std::cout << "\nYou: ";
        if (!std::getline(std::cin, user_input)) {
            break; 
        }

        if (user_input == "exit" || user_input == "quit") {
            speak("Exiting now. Goodbye.");
            break;
        }

        if (user_input.empty()) {
            continue;
        }

        std::string ai_response = get_ai_response(user_input, system_prompt);

        std::cout << "Siri-C: " << ai_response << std::endl;

        speak(ai_response);
    }
}

int main() {
    run_cpp_assistant();
    return 0;
}

File: AI/assistant.py
import os
import sys
import time

def generate_content(prompt):
    if "time" in prompt.lower():
        return f"The current time is {time.strftime('%I:%M %p')}."
    elif "joke" in prompt.lower():
        return "Why don't scientists trust atoms? Because they make up everything!"
    else:
        return f"I see. You asked about '{prompt}'. That is a complex topic that I can analyze for you."

def text_to_speech(text):
    print(f"\n[AI Speaking: '{text}']")
    time.sleep(len(text) * 0.05) 
    
def run_assistant():
    print("--- Jarvis Assistant Activated (Python) ---")
    print("Type 'exit' or 'quit' to stop the assistant.")
    
    system_instruction = "You are a friendly, concise, and helpful AI assistant named Jarvis. Keep your answers brief and conversational."
    
    chat_history = [
        {"role": "system", "content": system_instruction}
    ]

    text_to_speech("Hello, I am Jarvis. How can I assist you today?")

    while True:
        try:
            user_input = input("\nYou: ")
            
            if user_input.lower() in ["exit", "quit"]:
                text_to_speech("Goodbye. Have a great day!")
                break

            if not user_input.strip():
                continue

            chat_history.append({"role": "user", "content": user_input})
            
            ai_response = generate_content(user_input)
            
            chat_history.append({"role": "model", "content": ai_response})
            
            print(f"Jarvis: {ai_response}")
            text_to_speech(ai_response)

        except KeyboardInterrupt:
            print("\nShutting down...")
            text_to_speech("Shutting down now. See you soon!")
            break
        except Exception as e:
            error_message = f"An error occurred: {e}. Please try again."
            print(f"ERROR: {error_message}")
            text_to_speech("I encountered a problem. Please check the console for details.")
            
if __name__ == "__main__":
    run_assistant()

